src/transaction-runner.coffee

```diff
@@ -263,7 +277,17 @@ class TransactionRunner
       if transaction.test.valid == true
         if transaction.fail
           transaction.test.status = 'fail'
-          transaction.test.message = "Failed in after hook: " + transaction.fail
+
+          message = "Failed in after hook: " + transaction.fail
+          transaction.test.message = message
+
+          transaction['results'] ?= {}
+          transaction['results']['general'] ?= []
+
+          transaction['results']['general'].push {severity: 'error', message: message}
+
+          transaction['test']['results'] = transaction['results']
```

Comment:
inconsistent "copy" of results compared to the previous comment https://github.com/apiaryio/dredd/pull/222/files#diff-ecda976d2e1aba1fa8512ff2c6e505d7R83